### PR TITLE
Add multi-ldap login module

### PR DESCRIPTION
**Pull request by fviale**
The ldap module used is associated with a user domain.
Different domains will use different ldap configuration

Also, ensure that domain information is properly read and propagated through the server and nodes.

### CODE REVIEW

**Review by aminelouati — APPROVED** [review]

**Review by mboussaa — APPROVED** [review]

File: common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java [hunk 1]
```diff
@@ -51,13 +53,32 @@ public class ConnectionInfo implements Serializable {
     /**
      * @param url            the REST server URL
```
Comment (nebilbenmabrouk) [inline common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java:54]: the REST server URL => LDAP server URL

File: common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java [hunk 2]
```diff
@@ -102,4 +131,40 @@ public void setCredentialFile(File credentialFile) {
...
+     * @param fullLogin the login to parse
+     * @return the domain name, null if no domain is specified
+     * @since Scheduling 3.0.1
+     */
+    public static String parseDomain(String fullLogin) {
+        if (fullLogin.contains("\\")) {
+            String domain = fullLogin.substring(0, fullLogin.indexOf("\\"));
+            if ("".equals(domain.trim())) {
+                return null;
+            }
```
Comment (nebilbenmabrouk) [inline common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java:148]: This block is useless. It is included in the instruction:
`return "".equals(domain.trim()) ? null : domain;`

File: config/rm/settings.ini [hunk 3]
```diff
@@ -194,6 +194,11 @@ pa.rm.auth.pubkey.path=config/authentication/keys/pub.key
 # else, (if the path is absolute) it is directly interpreted
 pa.rm.ldap.config.path=config/authentication/ldap.cfg
 
+# Multi-domain LDAP configuration
+# a comma-separated list of domain_name:configuration_path
```
Comment (nebilbenmabrouk) [inline config/rm/settings.ini:198]: a comma-separated list of  **the pair** _domain_name:configuration_path_

File: common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java (hunk 1 above)
Comment (fviale) [reply common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java:54]: no, here it is the rest server url

File: common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java (hunk 2 above)
Comment (fviale) [reply common/common-api/src/main/java/org/ow2/proactive/authentication/ConnectionInfo.java:148]: you're right, this code is a blind copy/paste

**Pull request closed without merging**